src/lib/runtime/Web3FunctionRunner.ts

```diff
@@ -322,6 +324,39 @@ export class Web3FunctionRunner {
     }, 100);
   }
 
+  private _validateResult(result: Web3FunctionResult) {
+    if (!Object.keys(result).includes("canExec"))
+      throw new Error(
+        `Web3Function must return {canExec: bool, callData: string} or {canExec: bool, message: string}. Instead returned: ${JSON.stringify(
+          result
+        )}`
+      );
+
+    if (result.canExec) {
+      if (!Object.keys(result).includes("callData"))
+        throw new Error(
+          `Web3Function must return {canExec: bool, callData: string}. Instead returned: ${JSON.stringify(
+            result
+          )}`
+        );
+
+      if (result.callData.length < 10 || result.callData.slice(0, 2) !== "0x") {
+        throw new Error(
+          `Web3Function returned invalid callData. Returned: ${JSON.stringify(
+            result
+          )}`
+        );
+      }
+    }
+
+    if (!result.canExec && !Object.keys(result).includes("message"))
+      throw new Error(
+        `Web3Function must return {canExec: bool, message: string}. Instead returned: ${JSON.stringify(
+          result
+        )}`
+      );
```

Comment:
I would omit this one, as the `message` should be optional

Reply:
[cce131b](https://github.com/gelatodigital/web3-functions-sdk/pull/23/commits/cce131b96c97236d4d70878c4f209b73b7b36c8c)